server/container_create.go

```diff
@@ -109,54 +113,35 @@ func buildOCIProcessArgs(containerKubeConfig *pb.ContainerConfig, imageOCIConfig
 		return nil, fmt.Errorf("empty image config for %s", containerKubeConfig.Image.Image)
 	}
 
-	// We got an OCI Image configuration.
-	// We will only use it if the kubelet information is incomplete.
-
-	// First we set the process entry point.
-	if kubeCommands != nil {
-		// The kubelet command slice is prioritized.
-		processEntryPoint = kubeCommands
-	} else {
-		// Here the kubelet command slice is empty but
-		// we know that our OCI Image configuration is not empty.
-		// If the OCI image config has an ENTRYPOINT we use it as
-		// our process command.
-		// Otherwise we use the CMD slice if it's not empty.
-		if imageOCIConfig.Config.Entrypoint != nil {
-			processEntryPoint = imageOCIConfig.Config.Entrypoint
-		} else if imageOCIConfig.Config.Cmd != nil {
-			processEntryPoint = imageOCIConfig.Config.Cmd
+	// merge image config and kube config
+	// same as docker does today...
+	if imageOCIConfig != nil {
```

Comment:
nitpick: Now you can get rid of that check because you know for sure that at this point `imageOCIConfig` is not nil.